biogeophys/FatesHydroWTFMod.F90

```diff
@@ -800,22 +831,32 @@ subroutine set_wrf_param_tfs(this,params_in)
 
     class(wrf_type_tfs)  :: this
     real(r8), intent(in) :: params_in(:)
+    real(r8) :: th_max
+    real(r8) :: th_min
 
     this%th_sat   = params_in(1)
     this%th_res   = params_in(2)
     this%pinot    = params_in(3)
     this%epsil    = params_in(4)
     this%rwc_ft   = params_in(5)
-    this%cap_corr = params_in(6)
-    this%cap_int  = params_in(7)
-    this%cap_slp  = params_in(8)
-    this%pmedia   = int(params_in(9))
+    this%psicap   = params_in(6)
+    this%cap_corr = params_in(7)
+    this%cap_int  = params_in(8)
+    this%cap_slp  = params_in(9)
+    this%pmedia   = int(params_in(10))
+
+    ! Set DERIVED constants
+    ! used for interpolating in extreme ranges
+    th_max=max_sf_interp*(this%th_sat-this%th_res)+this%th_res-1.e-9_r8
+    th_min=min_sf_interp*(this%th_sat-this%th_res)+this%th_res+1.e-9_r8
+    this%psi_max     = this%psi_from_th(th_max)
+    this%dpsidth_max = this%dpsidth_from_th(th_max)
+    this%psi_min     = this%psi_from_th(th_min)
+    this%dpsidth_min = this%dpsidth_from_th(th_min)
+
 
-    call this%set_min_max(this%th_res,this%th_sat)
```

Comment:
I agree with @xuchongang , lets keep the ability to use extrapolations, and use a switch to turn that option on/off.  So lets keep this call to set those boundaries

Reply:
omitted removal of extrapolations, resolved